pyne/mesh.py

```diff
@@ -412,17 +412,21 @@ def __setitem__(self, key, value):
             mtag[i_ve[1]] = value if tsize == 1 else value[0]
         elif isinstance(key, slice):
             key = list(miter)[key]
-            v = np.empty(len(key), self.tag.type) if tsize == 1 else \
-                np.empty((len(key), tsize), self.tag.type)
+            if tsize == 1 and len(value.shape) == 1:
```

Comment:
Could you use the same trick as in #965 and just reshape this if the size is 1?
```
v = np.empty((len(key), tsize), self.tag.type)
if  tsize == 1 and len(value.shape) == 1:
    v.shape = (len(key), )
```

Reply:
Codes are changed here according to your suggestion. @gonuke